### ISSUE TITLE
The ability to keep the previous version of the secret when the google_secret_manager_secret_version is  being replaced

### ISSUE THREAD

**Issue by alexku7** [hashicorp/terraform-provider-google#14876]
Sometimes ( usually!) when the current version of the secret is rotated  we prefer not to destroy the replaced secret version .

It could be very useful to add a flag to google_secret_manager_secret_version resource which will specefy when I want to destroy/keep/disable the previous secret version when the google_secret_manager_secret_version is need to be replaced

**Comment by rileykarson** [comment]
I think we'd expect the secret to get replaced here- you're updating a specific version to a specific value. Adding a new google_secret_manager_secret_version resource is how you would add a new version of the secret while preserving the old, one.

Not entirely related, but possibly relevant: https://github.com/hashicorp/terraform-provider-google/issues/14488

**Comment by alexku7** [comment]
Sorry but I disagree.

Think about the following scenario
For example 
I use acme_certificate to renew  a certificate and store the renewed certificate in the google secret.
So the google_secret_manager_secret_version gets the secret_data as the output from the acme_certificate resource.

When I renew the new secret_data is stored, usually I want not to destroy the current secret version with the current certificate.  I might want to keep it as is or disable the version but not to destroy. 
It can be critical to rollback to the previous certificate in case of any  unexpected issue.

Bottom line, not always we want to destroy the previous version when we add a new version to the secret.

This is why the google secret has versions, no ?

**Comment by rileykarson** [comment]
The issue is that the `google_secret_manager_secret_version` manages a specific version of the secret- if the configuration you've written tells Terraform to delete the version, deleting the version *is* appropriate. That's where https://github.com/hashicorp/terraform-provider-google/issues/14488 is relevant, I expect that some (most?) users don't want secret versions managed with this level of granularity by Terraform.

**Comment by alexku7** [comment]
@rileykarson 
I am pretty sure
It's much simpler.

All what's needed is to add additional param like "dont_destroy_previous_version_on_change".

That's all.
If the secret_data param changed, the tf will create a new version with new secret_data but without destroying the current last version.

**Comment by rileykarson** [comment]
A `deletion_policy` field could work, yeah: https://googlecloudplatform.github.io/magic-modules/best-practices/#deletion-policy

**Comment by alexku7** [comment]
Thanks
